Allow approvers to impersonate the Cloud Build user-specified Service Account.

pasha-gh opened the pull request:
Fix #1064

pasha-gh commented:
Please ignore the long commit history left from previous changes.

xingao267 commented on templates/tfengine/components/cicd/main.tf:
```diff
@@ -134,6 +134,22 @@ resource "google_project_iam_member" "cloudbuild_builds_editors" {
+    "{{.}}",
+    {{- end}}
+  ])
+  service_account_id = google_service_account.cloudbuild_sa.name
```
Should use `local.cloudbuild_sa_email` here instead.

pasha-gh replied:
Fixed

xingao267 commented on templates/tfengine/components/cicd/main.tf:
```diff
@@ -134,6 +134,22 @@ resource "google_project_iam_member" "cloudbuild_builds_editors" {
   ]
 }
+
+# IAM permission to allow approvers to trigger builds.
```
IAM permission to allow approvers to impersonate the Cloud Build user-specified Service Account.

pasha-gh replied:
Updated.

xingao267 commented on templates/tfengine/components/cicd/main.tf:
```diff
@@ -134,6 +134,22 @@ resource "google_project_iam_member" "cloudbuild_builds_editors" {
+  member             = each.value
+  depends_on = [
+    google_project_service.services,
+    google_service_account.cloudbuild_sa
```
not needed

pasha-gh replied:
done.

xingao267 commented:
The PR title is not descriptive. Please update.

xingao267 commented:
@thomasfung-hk please take a look as well.

xingao267 commented:
Have you tested this?

xingao267 commented:
LGTM as well. I'll approve for merging once it's tested and verified.

thomasfung-hk commented:
Just realized that the integration test hasn't been run; should that be done first?

pasha-gh commented:
> Have you tested this?

Yes, I did test it with `google_service_account.cloudbuild_sa.name` and confirmed that build_editors have role/serviceAccount.user. We shouldn't have changed it to the email since `service_account_id` doesn't accept it.
Updated the PR and added `google_service_account.cloudbuild_sa.name` to the list of locals

xingao267 commented on templates/tfengine/components/cicd/main.tf:
```diff
@@ -41,8 +41,10 @@ data "google_project" "devops" {
+  cloudbuild_sa_name = "projects/${var.project_id}/serviceAccounts/${var.service_account}@${var.project_id}.iam.gserviceaccount.com"
 {{- else}}
   cloudbuild_sa_email = google_service_account.cloudbuild_sa.email
+  cloudbuild_sa_name = google_service_account.cloudbuild_sa.name
```
Is `cloudbuild_sa_id` more accurate here?
